lib/Doctrine/ORM/Mapping/ReflectionEnumProperty.php

```diff
@@ -63,12 +63,12 @@ public function getValue($object = null)
     }
 
     /**
-     * @param object                         $object
-     * @param int|string|int[]|string[]|null $value
+     * @param object                                    $object
+     * @param int|string|int[]|string[]|BackedEnum|null $value
      */
     public function setValue($object, $value = null): void
     {
-        if ($value !== null) {
+        if ($value !== null && ! $value instanceof BackedEnum) {
```

Comment:
```php
/**
 * @param object     $object
 * @param int|string|BackedEnum $value
 */
private function initializeEnumValue($object, $value): BackedEnum
{
    if ($value instanceof BackedEnum) {
        return $value;
    }

    $enumType = $this->enumType;
    ...
```

Reply:
Good point, `$value` can be an array of enums, fixed.